src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,66 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { FormEvent, useEffect, useState } from 'react';
+import { CommentData } from '../types/Comment';
+import { Post } from '../types/Post';
+
+type Props = {
+  handleOnAdd: (newComment: CommentData) => void,
+  isLoadingNewComment: boolean,
+  postSelected: Post | null,
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  handleOnAdd,
+  isLoadingNewComment,
+  postSelected,
+}) => {
+  const [name, setName] = useState('');
+  const [isNameError, setIsNameError] = useState(false);
+  const [email, setEmail] = useState('');
+  const [isEmailError, setIsEmailError] = useState(false);
+  const [body, setBody] = useState('');
+  const [isBodyError, setIsBodyError] = useState(false);
+
+  const handleReset = () => {
+    setIsBodyError(false);
+    setIsEmailError(false);
+    setIsNameError(false);
+    setName('');
+    setEmail('');
+    setBody('');
+  };
+
+  useEffect(() => {
+    handleReset();
+  }, [postSelected]);
+
+  const handleAdd = (event: FormEvent) => {
+    event.preventDefault();
+
+    const isNameEmpty = name.length === 0;
+    const isEmailEmpty = email.length === 0;
+    const isBodyEmpty = body.length === 0;
+
+    setIsNameError(isNameEmpty);
+    setIsEmailError(isEmailEmpty);
+    setIsBodyError(isBodyEmpty);
+
+    if (isNameEmpty || isEmailEmpty || isBodyEmpty) {
+      return;
+    }
+
+    const newComment: CommentData = {
+      name,
+      email,
+      body,
+    };
+
+    handleOnAdd(newComment);
+    setBody('');
+  };
 
-export const NewCommentForm: React.FC = () => {
   return (
-    <form data-cy="NewCommentForm">
+    <form data-cy="NewCommentForm" onSubmit={(e) => handleAdd(e)}>
```

Comment:
I can not avoid condition field on leangth, because user have oppotunity press button (add comment) anywhere, but I improve, directly input field `setName(target.value.trimStart());`